src/java/htsjdk/samtools/metrics/MetricBase.java

```diff
@@ -84,7 +85,10 @@ public int hashCode() {
         int result = 0;
         for (final Field f : getClass().getFields()) {
             try {
-                result = 31 * result + f.get(this).hashCode();
+                Object value = f.get(this);
+                value = new FormatUtil().format(value); //format the value the way it will be written to disk
```

Comment:
If equality is based on the formatted value, hashCode will be inconsistent with equals if it is not also based on the formatted value.  I.e. `hashCode(1.0000000001) ~= hashCode(1) even though 1.000000001 == 1.

I believe that simplicity should take preference over performance here since these are not very performance critical classes.  The earlier version had explicit checks to only perform the formatting if it was decimal value that could be formatted, but that seemed confusing and brittle.